File: api-servidor/src/domain/entities/animal.ts
import { AnimalSpecies } from '../enums/animal-species';
import { AnimalGender } from '../enums/animal-gender';

export interface AnimalProps {
  id?: string;
  tutorId: string;
  name: string;
  species: AnimalSpecies;
  breed?: string | null;
  gender: AnimalGender;
  birthDate?: Date | null;
  weight?: number | null;
  color?: string | null;
  microchipNumber?: string | null;
  photoUrl?: string | null;
  allergies?: string | null;
  notes?: string | null;
  createdAt?: Date;
  updatedAt?: Date;
  deletedAt?: Date | null;
}

export class Animal {
  public readonly id?: string;
  public tutorId: string;
  public name: string;
  public species: AnimalSpecies;
  public breed?: string | null;
  public gender: AnimalGender;
  public birthDate?: Date | null;
  public weight?: number | null;
  public color?: string | null;
  public microchipNumber?: string | null;
  public photoUrl?: string | null;
  public allergies?: string | null;
  public notes?: string | null;
  public readonly createdAt?: Date;
  public readonly updatedAt?: Date;
  public readonly deletedAt?: Date | null;

  constructor(props: AnimalProps) {
    this.id = props.id;
    this.tutorId = props.tutorId;
    this.name = props.name;
    this.species = props.species;
    this.breed = props.breed;
    this.gender = props.gender;
    this.birthDate = props.birthDate;
    this.weight = props.weight;
    this.color = props.color;
    this.microchipNumber = props.microchipNumber;
    this.photoUrl = props.photoUrl;
    this.allergies = props.allergies;
    this.notes = props.notes;
    this.createdAt = props.createdAt;
    this.updatedAt = props.updatedAt;
    this.deletedAt = props.deletedAt;
  }
}

File: mobile/lib/features/animal/domain/entities/animal.dart
import 'package:equatable/equatable.dart';
import '../../../../core/domain/enums/animal_species.dart';
import '../../../../core/domain/enums/animal_gender.dart';

class Animal extends Equatable {
  final String? id;
  final String tutorId;
  final String name;
  final AnimalSpecies species;
  final String? breed;
  final AnimalGender gender;
  final DateTime? birthDate;
  final double? weight;
  final String? color;
  final String? microchipNumber;
  final String? photoUrl;
  final String? allergies;
  final String? notes;
  final bool isActive;
  final DateTime? createdAt;

  const Animal({
    this.id,
    required this.tutorId,
    required this.name,
    required this.species,
    this.breed,
    required this.gender,
    this.birthDate,
    this.weight,
    this.color,
    this.microchipNumber,
    this.photoUrl,
    this.allergies,
    this.notes,
    this.isActive = true,
    this.createdAt,
  });

  int? get ageInMonths {
    if (birthDate == null) return null;
    final now = DateTime.now();
    return (now.year - birthDate!.year) * 12 + (now.month - birthDate!.month);
  }

  String? get ageDisplayText {
    final months = ageInMonths;
    if (months == null) return null;
    if (months < 12) return '$months meses';
    final years = months ~/ 12;
    final remainingMonths = months % 12;
    if (remainingMonths == 0) return '$years ano${years > 1 ? 's' : ''}';
    return '$years ano${years > 1 ? 's' : ''} e $remainingMonths mes${remainingMonths > 1 ? 'es' : ''}';
  }

  @override
  List<Object?> get props => [id, tutorId, name];
}
